models/migrations/v24.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2017 Gitea. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"github.com/go-xorm/xorm"
+)
+
+type StatusTable struct {
+	ID          int64  `xorm:"pk autoincr"`
+	Index       int64  `xorm:"INDEX UNIQUE(repo_sha_index)"`
+	RepoID      int64  `xorm:"INDEX UNIQUE(repo_sha_index)"`
+	State       string `xorm:"TEXT NOT NULL"`
+	SHA         string `xorm:"TEXT NOT NULL INDEX UNIQUE(repo_sha_index)"`
```

Comment:
Since there's talk about replacing SHA1 in git with "something else" we probably shouldn't make assumptions about it's length.